util/io/src/main/scala/sbt/IO.scala

```diff
@@ -68,18 +86,29 @@ object IO {
   }
```

Comment:
Looks to me like it used to fallback to using the system classloader (via old `classLocation(cl: Class[_]): URL`) and now it doesn't.